### PR TITLE
Add private map count to /me

### PR DESCRIPTION
**Pull request by gonzaloriestra**
Closes https://github.com/CartoDB/cartodb/issues/15463

Also, now Kuviz visualizations are taken into account for usage metrics (they were recently included in the quotas).

### CODE REVIEW

**Comment by alrocar** [comment]
@esloho Could you take a look when you have time, please? 🙏

File: spec/queries/carto/visualization_query_builder_spec.rb [hunk 1]
```diff
@@ -505,4 +505,60 @@ def preload_activerecord_metadata
...
+    end
+
+    after(:all) do
+      @user.destroy
+    end
+
+    it 'returns the private maps of a user' do
+      result = Carto::VisualizationQueryBuilder.user_private_privacy_visualizations(@user).build
+
+      expect(result.count).to eq 1
```
Comment (esloho) [inline spec/queries/carto/visualization_query_builder_spec.rb]: I've seen using in tests 2 types of assertions:
- expect(result.count).to eq 1
- result.count.should eq 1

I find the second one less verbose and easy to read (also my love for spock tests grows deep on me xD) but I don't know if there's any difference beyond aesthetics (performance, behavior...)?

**Comment by esloho** [comment]
Let me know if I should do the acceptance as well :)

File: app/models/carto/visualization.rb [hunk 2]
```diff
@@ -41,6 +41,7 @@ class Carto::Visualization < ActiveRecord::Base
   TYPE_KUVIZ = 'kuviz'.freeze
 
   VALID_TYPES = [TYPE_CANONICAL, TYPE_DERIVED, TYPE_SLIDE, TYPE_REMOTE, TYPE_KUVIZ].freeze
+  MAP_TYPES = [TYPE_DERIVED, TYPE_KUVIZ].freeze
```
[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply app/models/carto/visualization.rb]: - `TYPE_CANONICAL = 'table'.freeze` => builder datasets
- `TYPE_DERIVED = 'derived'.freeze` => builder maps
- `TYPE_SLIDE = 'slide'.freeze` => deprecated, powerpoint-like visualizations
- `TYPE_REMOTE = 'remote'.freeze` => data library (common data)
- `TYPE_KUVIZ = 'kuviz'.freeze` => custom visualizations (CARTOframes)

File: spec/queries/carto/visualization_query_builder_spec.rb (hunk 1 above)
Comment (gonzaloriestra) [reply spec/queries/carto/visualization_query_builder_spec.rb]: There's no difference, `should` is the old way, `expect` the new and recommended one by RSpec.

**Comment by gonzaloriestra** [comment]
@esloho thanks! Don't worry about the acceptance, @csubira has to integrate this and I guess she can test it then :pray:

File: spec/queries/carto/visualization_query_builder_spec.rb (hunk 1 above)
Comment (esloho) [reply spec/queries/carto/visualization_query_builder_spec.rb]: So I'm being _old_, so much like me :D  Thanks!

**Comment by csubira** [comment]
Thanks @esloho and @gonzaloriestra! Acceptance done and it works great! Ready to be merged so we can use it for the rest of the free plan implementation!

**Comment by alrocar** [comment]
I'm going to merge it

**Pull request merged**